GoogleForCreators/web-stories-wp
# Performance: Optimize checklist rendering

## Context
Part of tightening up perf of the editor for primary actions like DnD

## Summary
Removes unneeded rendering of elements within the checklist when the checklist is closed

<img width="1434" alt="Screen Shot 2021-07-30 at 1 18 58 PM" src="https://user-images.githubusercontent.com/35983235/127697585-e5425fe1-ea54-45af-889a-ef6a914564cf.png">
<img width="1267" alt="Screen Shot 2021-07-30 at 1 18 42 PM" src="https://user-images.githubusercontent.com/35983235/127697590-d35c2465-c380-4cfc-b89b-f67453753bf6.png">


## Relevant Technical Choices
just used a provider and transition group callbacks to see when the component was mounted and pass that mounted value around.


## To-do
NA


## User-facing changes
NA


## Testing Instructions
Open up the editor help center and see that there were no regressions. Can inspect react flame chart and see that it takes up less time to render when it's closed as it did before or just check rendered dom elements and see that there are no rendered children of the popup when it's closed.


### QA






This PR can be tested by following these steps:

1.

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #8446
